src/main/java/com/amazon/ion/impl/IonCursorBinary.java

```diff
@@ -3361,6 +3361,10 @@ private void setCheckpointAfterValueHeader() {
             case START_CONTAINER:
                 setCheckpoint(CheckpointLocation.AFTER_CONTAINER_HEADER);
                 break;
+            case NEEDS_INSTRUCTION:
+                // A macro invocation header has just been read.
+                setCheckpoint(CheckpointLocation.BEFORE_UNANNOTATED_TYPE_ID);
```

Comment:
This moves the cursor past the macro invocation header, positioning it on the first argument (if any). This method is called when parsing expression groups.